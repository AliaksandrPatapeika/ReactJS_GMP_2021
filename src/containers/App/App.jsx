import './App.less';

import React from 'react';
import {Provider} from 'react-redux';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';

import store from '../../store';
import ErrorBoundary from '../ErrorBoundary';
import HomePage from '../HomePage';
import MovieDetailsPage from '../MovieDetailsPage';
import PageNotFound from '../PageNotFound';
import SearchPage from '../SearchPage';

const App = () => (
  <Provider store={store}>
    <ErrorBoundary>
      <Router>
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/search" component={SearchPage} />
          <Route exact path="/film/:id" component={MovieDetailsPage} />
          <Route path="*" component={PageNotFound} />
        </Switch>
      </Router>
    </ErrorBoundary>
  </Provider>
);

export default App;
