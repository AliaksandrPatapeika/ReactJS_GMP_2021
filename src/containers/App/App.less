@import "../../less/reset";
@import "../../less/variables";
@import "../../less/common";

body {
	background-color: @gray;

	/* Sticky Footer with Flexbox */
	#app {
		display: flex;
		flex-direction: column;
		height: 100%;

		.mainContainer,
		.modalWindowContainer {
			flex: 1 0 auto;
		}

		.footerContainer {
			flex-shrink: 0;
		}
	}
}