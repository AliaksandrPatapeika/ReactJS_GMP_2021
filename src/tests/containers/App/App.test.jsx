import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../../containers/App';

describe('App', () => {
  test('renders correctly', () => {
    const app = renderer
      .create(<App />)
      .toJSON();

    expect(app).toMatchSnapshot();
  });
});
