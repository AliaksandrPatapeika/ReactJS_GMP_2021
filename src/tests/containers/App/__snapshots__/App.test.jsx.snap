// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <header
    className="headerContainer"
  >
    <div
      className="headerBlurImageLayer"
    >
      <div
        className="headerTopContainer"
      >
        <div
          className="logoContainer"
        >
          <span
            className="netflix"
          >
            netflix
          </span>
          <span
            className="roulette"
          >
            roulette
          </span>
        </div>
        <button
          className="AddMovieButton"
          type="button"
        >
          + ADD MOVIE
        </button>
      </div>
      <div
        className="headerMainContainer"
      >
        <h1
          className="headerTitle"
        >
          FIND YOUR MOVIE
        </h1>
        <div
          className="searchContainer"
        >
          <input
            className="searchInput"
            placeholder="What do you want to watch?"
          />
          <button
            className="searchButton"
            type="button"
          >
            SEARCH
          </button>
        </div>
      </div>
    </div>
  </header>,
  <main
    className="mainContainer"
  >
    <div
      className="menuContainer"
    >
      <nav>
        <ul
          className="resultsFilterContainer"
        >
          <li
            className="filterItem active"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
          >
            ALL
          </li>
          <li
            className="filterItem"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
          >
            DOCUMENTARY
          </li>
          <li
            className="filterItem"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
          >
            COMEDY
          </li>
          <li
            className="filterItem"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
          >
            HORROR
          </li>
          <li
            className="filterItem"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
          >
            CRIME
          </li>
        </ul>
      </nav>
      <div
        className="resultsSortContainer"
      >
        <div
          className="sortBy"
        >
          SORT BY
        </div>
        <div
          className="releaseDate"
        >
          RELEASE DATE
        </div>
        <div
          aria-label="Sort Button"
          className="arrow arrow-down"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        />
      </div>
    </div>
    <div
      className="resultCountContainer"
    >
      <span
        className="resultCount"
      >
        39
         
      </span>
      <span>
        movies found
      </span>
    </div>
    <div
      className="moviesListContainer"
    >
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Pulp Fiction
          </div>
          <div
            className="itemReleaseDate"
          >
            2004
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Action & Adventure
        </div>
      </div>
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Bohemian Rhapsody
          </div>
          <div
            className="itemReleaseDate"
          >
            2003
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Drama, Biography, Music
        </div>
      </div>
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Kill Bill: Vol 2
          </div>
          <div
            className="itemReleaseDate"
          >
            1994
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Oscar winning Movie
        </div>
      </div>
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Avengers: War of Infinity
          </div>
          <div
            className="itemReleaseDate"
          >
            2004
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Action & Adventure
        </div>
      </div>
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Inception
          </div>
          <div
            className="itemReleaseDate"
          >
            2003
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Action & Adventure
        </div>
      </div>
      <div
        className="movieCardContainer"
      >
        <img
          alt="Item"
          className="itemImage"
          src="test-file-stub"
        />
        <button
          className="threeDotsIcon"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="MovieButton"
            src="test-file-stub"
          />
        </button>
        <div
          className="descriptionContainer"
        >
          <div
            className="itemTitle"
          >
            Reservoir dogs
          </div>
          <div
            className="itemReleaseDate"
          >
            1994
          </div>
        </div>
        <div
          className="itemGenre"
        >
          Oscar winning Movie
        </div>
      </div>
    </div>
  </main>,
  <footer
    className="footerContainer"
  >
    <div
      className="logoContainer"
    >
      <span
        className="netflix"
      >
        netflix
      </span>
      <span
        className="roulette"
      >
        roulette
      </span>
    </div>
  </footer>,
]
`;
